packages/clay-date-picker/src/index.tsx

```diff
@@ -226,7 +245,23 @@ const ClayDatePicker: React.FunctionComponent<IProps> = React.forwardRef<
 		/**
 		 * Flag to indicate if date is expanded.
 		 */
-		const [expanded, setExpanded] = React.useState(initialExpanded);
+		const [internalExpanded, setInternalExpanded] = React.useState(
+			expanded || initialExpanded
+		);
+
+		const setExpanded = (val: boolean) => {
+			if (expanded !== val && onExpandedChange) {
+				onExpandedChange(val);
+			}
+
+			setInternalExpanded(val);
+		};
+
+		React.useEffect(() => {
+			if (expanded !== undefined) {
+				setInternalExpanded(expanded);
+			}
+		}, [expanded]);
```

Comment:
Perhaps we can decrease the use here of `useEffect` which could happen a double rendering due to the `useState` that the developer will have to use to control. I'm thinking of something like this:

```js
const expanded = externalExpanded !== undefined ? externalExpanded : internalExpanded;
const setExpanded = onExpandedChange ? onExpandedChange : setInternalExpanded;
```
`externalExpanded` can only be an alias of `expanded` prop. What do you think? it's makes sense for you?

Reply:
@matuzalemsteles let me try that out. That was my goal but my first iteration was having issues. Let me look again to see if this is possible.